tsdb/head.go

```diff
@@ -867,8 +860,8 @@ func (h *Head) appender(appendID, cleanupAppendIDsBelow uint64) *headAppender {
 		maxt:                  math.MinInt64,
 		samples:               h.getAppendBuffer(),
 		sampleSeries:          h.getSeriesBuffer(),
-		appendID:              appendID,
-		cleanupAppendIDsBelow: cleanupAppendIDsBelow,
+		appendID:              h.iso.newAppendID(),
+		cleanupAppendIDsBelow: h.iso.lowWatermark(),
```

Comment:
Note: go will go execute that in correct order: https://play.golang.org/p/tjCrTI77UwZ

Reply:
Is that guaranteed by the spec?

Reply:
I don't find that info, and for maps and slices it is not guaranteed. https://www.facebook.com/ I have asked golang-nuts.

Reply:
In the meantime I have made the execution order explicit.